scripts/merge_defining_mutations.py

```diff
@@ -0,0 +1,355 @@
+    return y
+
+
+def process_cornelius_file(path):
+    with open(path) as f:
+        data = json.load(f)
+
+    raw_lineages = [mut for mut in data.values()]
+
+    df = pl.from_records(raw_lineages)
+
+    rename = df.rename(
+        {'nextstrainClade': 'nextstrain_clade',
+         'designationDate': 'designation_date',
+         'nucSubstitutions': 'nuc_sub_wuhan',
+         'nucDeletions': 'nuc_del_wuhan',
+         'nucSubstitutionsNew': 'nuc_sub_pango_parent',
+         'nucDeletionsNew': 'nuc_del_pango_parent',
+         'aaSubstitutions': 'aa_sub_wuhan',
+         'aaDeletions': 'aa_del_wuhan',
+         'aaSubstitutionsNew': 'aa_sub_pango_parent',
+         'aaDeletionsNew': 'aa_del_pango_parent'}
+    )
+
+    lineages = rename.select(
+        pl.col('lineage'),
+        pl.col('unaliased'),
+        pl.col('parent'),
+        pl.col('children'),
+        pl.col('nextstrain_clade'),
+        pl.col('designation_date'))
+
+    mutations = rename.select(
+        'lineage',
+        'nextstrain_clade',
+        'nuc_sub_wuhan',
+        'nuc_del_wuhan',
+        'nuc_sub_pango_parent',
+        'nuc_del_pango_parent',
+        'aa_sub_wuhan',
+        'aa_del_wuhan',
+        'aa_sub_pango_parent',
+        'aa_del_pango_parent'
+    )
+
+    empty_strings_removed = mutations.to_pandas().applymap(remove_empty_strings)
+
+    deletion_columns = ['nuc_del_wuhan', 'nuc_del_pango_parent']
+    split_deletions = empty_strings_removed[deletion_columns].applymap(
+        lambda x: [[int(pos) for pos in del_range.split('-')] for del_range in x])
+    deletions_to_ranges = split_deletions.applymap(
+        lambda x: [list(range(*del_range)) if len(del_range) > 1 else del_range for del_range in x])
+    flatten_deletions = deletions_to_ranges.applymap(lambda x: list(pd.core.common.flatten(x)))
+    expand_deletions = empty_strings_removed.copy()
+    expand_deletions[deletion_columns] = flatten_deletions
+
+    reformat_deletions = pl.from_pandas(expand_deletions).with_columns(
+        pl.col(col_name).list.eval(
+            pl.concat_str(
+                pl.lit('X'),
+                pl.element(),
+                pl.lit('-'))
+        ) for col_name in deletion_columns
+    )
+
+    combine_nuc_changes = reformat_deletions.with_columns(
+        nuc_change_wuhan=pl.concat_list(
+            'nuc_sub_wuhan',
+            'nuc_del_wuhan'),
+        nuc_change_pango_parent=pl.concat_list(
+            'nuc_sub_pango_parent',
+            'nuc_del_pango_parent'
+        ),
+        aa_change_wuhan=pl.concat_list(
+            'aa_sub_wuhan',
+            'aa_del_wuhan'
+        ),
+        aa_change_pango_parent=pl.concat_list(
+            'aa_sub_pango_parent',
+            'aa_del_pango_parent'
+        )
+    )
+
+    # TODO: make sure the nuc to aa matching is scientifically correct
+    match_aa_changes = pl.struct('nuc_change', 'aa_change').map_elements(
+        lambda x: match_nuc_to_aas(x['nuc_change'], x['aa_change']), pl.String
```

Comment:
```suggestion
        lambda row: match_nuc_to_aas(row['nuc_change'], row['aa_change']), pl.String
```
I'm not sure if this is a row, but it is a bit better than x